src/interfaces/interfaces.ts

```diff
@@ -42,6 +46,10 @@ namespace interfaces {
         clone(): T;
     }
 
+    export type BindingActivation<T> = (context: interfaces.Context, injectable: T) => T | Promise<T>;
```

Comment:
Just wondering, should we allow users to work directly with the context? There's no easy way to use it.

It gives access to the current resolution plan which is, at this point, partially resolved. It gives access to the request, which can be used to access all the children and parent requests. The update of these models could lead to an undefined behavior. For example, a user could technically create an infinite loop with a plan cycle. 

I think it's great to provide users a **read only** tool to traverse the dependency tree, but I wonder if `context` is the right tool for them.

What are your thoughts about this? I'd love to have this awesome feature merged, but I prefer to discuss this topic to be sure we provide a great tool with no side effects.

Reply:
Never mind, onActivation handlers already reveive a context